src/pudl/analysis/record_linkage/classify_plants_ferc1.py

```diff
@@ -0,0 +1,359 @@
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "construction_year",
+                        "columns": ["construction_year"],
+                        "transformer": "category",
+                        "cleaning_ops": ["fix_int_na"],
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "utility_id_ferc1",
+                        "columns": ["utility_id_ferc1"],
+                        "transformer": "category",
+                    }
+                ),
+                ColumnTransform(
+                    **{
+                        "step_name": "fuel_fraction_mmbtu",
+                        "columns": fuel_cols,
+                        "transformer": Pipeline(
+                            [("scaler", MinMaxScaler()), ("norm", Normalizer())]
+                        ),
+                        "cleaning_ops": ["null_to_zero"],
+                    }
+                ),
+            ],
+        }
+    )
+
+
+def plants_steam_assign_plant_ids(
+    ferc1_steam_df: pd.DataFrame,
+    ferc1_fuel_df: pd.DataFrame,
+    fuel_categories: list[str],
+) -> pd.DataFrame:
+    """Assign IDs to the large steam plants."""
+    ###########################################################################
+    # FERC PLANT ID ASSIGNMENT
+    ###########################################################################
+    # Now we need to assign IDs to the large steam plants, since FERC doesn't
+    # do this for us.
+    logger.info("Identifying distinct large FERC plants for ID assignment.")
+
+    # Grab fuel consumption proportions for use in assigning plant IDs:
+    fuel_fractions = fuel_by_plant_ferc1(ferc1_fuel_df, fuel_categories)
+    ffc = list(fuel_fractions.filter(regex=".*_fraction_mmbtu$").columns)
+
+    ferc1_steam_df = ferc1_steam_df.merge(
+        fuel_fractions[["utility_id_ferc1", "plant_name_ferc1", "report_year"] + ffc],
+        on=["utility_id_ferc1", "plant_name_ferc1", "report_year"],
+        how="left",
+    )
+
+    fuel_cols = list(ferc1_steam_df.filter(regex=".*_fraction_mmbtu$").columns)
+
+    # Train the classifier using DEFAULT weights, parameters not listed here.
+    clf = construct_ferc1_plant_matching_model(fuel_cols)
+    ferc1_steam_df = clf.fit_predict(ferc1_steam_df)
+
+    # Set the construction year back to numeric because it is.
+    ferc1_steam_df["construction_year"] = pd.to_numeric(
+        ferc1_steam_df["construction_year"], errors="coerce"
+    )
+    # We don't actually want to save the fuel fractions in this table... they
+    # were only here to help us match up the plants.
+    ferc1_steam_df = ferc1_steam_df.drop(ffc, axis=1)
+    ferc1_steam_df = revert_filled_in_string_nulls(ferc1_steam_df)
+
+    return ferc1_steam_df
+
+
+def revert_filled_in_string_nulls(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
BTW I think I'm going to use these two "revert" functions for the FERC to EIA match, so it might be worth pulling them out into a more general module (`cleaning_steps`? not really a cleaning step but an uncleaning step)

Reply:
Good to know. I'm working on some refactoring/reorganization in the ferc-eia integration, so I'll probably leave that for the next PR